db/compaction/compaction_job.cc

```diff
@@ -380,6 +380,7 @@ void CompactionJob::ReleaseSubcompactionResources() {
   if (extra_num_subcompaction_threads_reserved_ == 0) {
     return;
   }
+  db_mutex_->Lock();
```

Comment:
Ditto

Reply:
Unfortunately I can't easily do InstrumentedMutexLock for this one cuz the function right after its unlock "ShrinkSubcompactionResources(extra_num_subcompaction_threads_reserved_);" but before the scope ends need to acquire lock again. So I won't be fixing this one.

Reply:
How about
```cpp
{
  InstrumentedMutexLock (db_mutex_);
  assert(...)
}
```

Furthermore, mutex lock-unlock is not needed if in opt mode, but if we differentiate between debug and opt mode, there may be a gap in test coverage, thus I am fine with current.

Reply:
Got you - yeah I can change to your suggested version! This is the easy way that I didn't think of!

Reply:
Fixed